### ISSUE TITLE
Possible improvement of sequential CTMRG

### ISSUE THREAD

**Issue by Yue-Zhengyuan** [QuantumKitHub/PEPSKit.jl#88]
This issue is only a suggestion and I may have some misunderstanding... Let me focus on the left move of sequential CTMRG:
```
    -----> 
    C1 - T1 -   r-1
    |    ‖
    T4 = M' =   r
    |    ‖
    C4 - T3 -   r+1
    c-1  c 
```
Currently, it appears that the enlarged NW/SW corners for all sites and projectors on all columns are found at once, and the renormalization of one column is independent of the renormalization of the previous column. However, I only handle one column each time (this is also the approach in the full update). After I update C1, T4, C4 in the (c-1)-th column, when finding the projectors for the c-th column, I will use the *updated* value of C1, T4, C4 in the (c-1)-th column. Naively thinking, this may help speed up convergence a little.

Is my understanding correct? If so, I would also like to know why you decided to use the current approach. Thanks!

**Comment by pbrehmer** [comment]
Thank you for raising this question! You are right, currently all rows and columns are renormalized independently in the `:sequential` mode. One advantage of this is that you can of course parallelize this step which might pay off for larger unit cells. However, in terms of convergence speed it might be advantageous to do what you suggest (and what is probably more conventional anyways): to enlarge and renormalize also the unit cell sequentially.

The reason why we decided to do this is probably because it was part of the old PEPSKit code and there doesn't seem to be an immediate obvious difference in the convergence speed between the `:simultaneous` and `:sequential` modes. That being said, we never really benchmarked the two modes against each other, so it is up for discussion if there are cases where `:sequential` significantly outperforms `:simultaneous` (i.e. where simultaneous renormalization of the unit cell is worse than a sequential renormalization).

Now that I think about it, I find your suggestion to be the better choice for `:sequential`. Let me open up a PR and implement it - it shouldn't be too much of an effort.

**Comment by pbrehmer** [comment]
Implemented in #90 and refactored in #99.
